### ISSUE TITLE
recursiveGet: switch on line 19 falls through to a guaranteed index out of range panic

### ISSUE THREAD

**Issue by eloff** [yasuyuky/jsonpath#1]
If we enter this switch, len(path) == 0. However, if none of the conditions match (e.g. data is an int) then it falls through to path[0] which will panic with index out of range. I recommend adding a default case to the switch or after the switch to return an informative error.

```
if len(path) == 0 {
	switch data.(type) {
	case string:
		return data.(string)
	case float64:
		return data.(float64)
	case bool:
		return data.(bool)
	case nil:
		return nil
	case []interface{}:
		return data
	case map[string]interface{}:
		return data
	}
}

switch path[0].(type) {
```

**Comment by yasuyuky** [comment]
>  (e.g. data is an int)

I think data can never be an int.

**Comment by eloff** [comment]
data was actually an int for me, and I modified the library as described to work around it. Perhaps I was using the library in a way not intended, but either way the fix won't do any harm.

**Comment by yasuyuky** [comment]
Do you have any example for this case?

**Comment by eloff** [comment]
No, I no longer have access to the codebase, and we switched away from this library in the end.

My gut feeling is the "object tree" in the interface{} was actually an int, and this was allowed by our data model but not by your library. Would that make sense?

**Comment by webbrain-one** [comment]
Thanks for flagging this! I opened https://github.com/yasuyuky/jsonpath/pull/3, which adds a default case for the `len(path) == 0` switch and returns an informative error instead of panicking when the data type isn't supported.
